nemo/core/classes/modelPT.py

```diff
@@ -606,10 +606,44 @@ def setup_optimizer_param_groups(self):
             See https://pytorch.org/docs/stable/optim.html for more information.
             By default, ModelPT will use self.parameters().
             Override this method to add custom param groups.
-    """
-        param_groups = None
-        if hasattr(self, 'parameters'):
-            param_groups = [{'params': self.parameters()}]
+            In the config file, add 'optim_param_groups' to support different LRs 
+            for different components (unspecified params will use the default LR):
+            model:
+                optim_param_groups:
+                    encoder: 1e-4
+                    decoder: 1e-3
+                optim:
+                    lr: 3e-3   
+        """
+        if not hasattr(self, "parameters"):
+            self._optimizer_param_groups = None
+            return
+
+        known_groups = []
+        param_groups = []
+        if hasattr(self.cfg, "optim_param_groups"):
+            param_groups_cfg = self.cfg.optim_param_groups
+            for group, lr in param_groups_cfg.items():
+                module = getattr(self, group, None)
```

Comment:
Check for none and raise error that invalid module was provided c

Reply:
Better to be explicit rather than silently be incorrect.

Reply:
Done